### PR TITLE
[Merged by Bors] - Refactoring: new LdapAuthenticationProvider framework code

### PR DESCRIPTION
**Pull request by fhennig**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by fhennig** [comment]
test run: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/59/

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,15 +4,18 @@
 
 ### Changed
 
-- Fixed the RoleGroup `selector`. It was not used before. ([#208])
+- `operator-rs` `0.25.2` -> `0.27.1` ([#197]).
 - `operator-rs` `0.27.1` -> `0.30.1` ([#208])
+- `operator-rs` `0.30.1` -> `0.23.0` ([#216]).
```
Comment (adwk67) [inline CHANGELOG.md:9]:
```suggestion
- `operator-rs` `0.30.1` -> `0.31.0` ([#216]).
```

**Review by adwk67 — APPROVED** [review]
lgtm!

File: rust/operator-binary/src/airflow_controller.rs [hunk 2]
```diff
@@ -803,74 +799,17 @@ fn add_authentication_volumes_and_volume_mounts(
...
-            }
-
+            ldap.add_volumes_and_mounts(pb, vec![cb]);
             Ok(())
         }
         _ => AuthenticationClassProviderNotSupportedSnafu {
             authentication_class_provider: authentication_class.spec.provider.to_string(),
-            authentication_class: ObjectRef::<AuthenticationClass>::new(authentication_class_name),
+            authentication_class: ObjectRef::<AuthenticationClass>::new(
+                authentication_class.metadata.name.as_ref().unwrap(),
```
Comment (sbernauer) [inline rust/operator-binary/src/airflow_controller.rs:810]: I *think* this should exist, you may need a trait import
```suggestion
                authentication_class.name_unchecked(),
```

**Review by sbernauer — APPROVED** [review]
Nice

**Comment by fhennig** [comment]
bors merge

File: rust/operator-binary/src/airflow_controller.rs (hunk 2 above)
Comment (fhennig) [reply rust/operator-binary/src/airflow_controller.rs]: I changed it :+1:

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**